docs/modules/plugin/partials/process-asciidoc-mojo-parameters.adoc

```diff
@@ -0,0 +1,203 @@
+====
+
+sourceDocumentName:: an override to process a single source file; defaults to all files in [.path]`$\{sourceDirectory}`.
+sourceDocumentExtensions:: (named `extensions` in v1.5.3 and below) a `List<String>` of non-standard file extensions to convert.
+Currently, _ad_, _adoc_, and _asciidoc_ will be converted by default
+resources:: list of resource files to copy to the output directory (e.g., images, css).
+The configuration follows the same patterns as the `maven-resources-plugin`.
+If not set, all resources inside `$\{sourceDirectory}` are copied.
++
+[NOTE]
+====
+Converters that embed resources such as images into the output document need to be able to locate those resources at conversion time.
+For example, when generating a PDF (or HTML with the `data-uri` attribute set), all images need to be aggregated under a common root (i.e., image catalog).
+The `imagesdir` attribute should be overridden to point to that folder.
+When converting to HTML, images must be copied to the output location so that the browser can resolve those images when the user views the page.
+====
++
+[source, xml]
+----
+<resources>
+    <resource>
+        <!-- (Mandatory) Directory to copy from. Paths are relative to maven's ${baseDir} -->
+        <directory>DIRECTORY</directory>
+        <!-- (Optional) Directory to copy to. By default uses the option `outputDirectory` -->
+        <targetPath>OUTPUT_DIR</targetPath>
+        <!-- (Optional) NOTE: SVN, GIT and other version control files are excluded by default, there's no need to add them -->
+        <excludes>
+            <exclude>**/.txt</exclude>
+        </excludes>
+        <!-- (Optional) If not set, includes all files but default exceptions mentioned -->
+        <includes>
+            <include>**/*.jpg</include>
+            <include>**/*.gif</include>
+        </includes>
+    </resource>
+    ...
+</resources>
+----
+outputDirectory:: locations where converted sources and copied resources will be places.
+Note that relative paths are added to the project root path.
+Defaults to [.path]_${project.build.directory}/generated-docs_.
+outputFile:: defaults to `null`, used to override the name of the generated output file, can be a relative or absolute path.
+Useful for backends that create a single file, e.g. the pdf backend.
+All output will be redirected to the same file, the same way as the `-o, --out-file=OUT_FILE` option from the `asciidoctor` CLI command.
+baseDir:: (not Maven's basedir) enables to set the root path for resources (e.g. included files), defaults to `$\{sourceDirectory}`
+skip:: set this to `true` to bypass generation, defaults to `false`
+preserveDirectories:: enables to specify whether the documents should be converted in the same folder structure as in the source directory or not, defaults to `false`.
+When `true`, instead of generating all output in a single folder, output files are generated in the same structure.
+See the following example
++
+[source]
```

Comment:
The style on a text-based diagram should be listing, not source (or you can make it a literal block).

Reply:
I am going to got for literal.